dogstatsd/src/datadog.rs

```diff
@@ -149,40 +147,19 @@ impl DdApi {
     }
 
     /// Ship a serialized series to the API, blocking
-    pub async fn ship_series(&self, series: &Series) {
-        let body = serde_json::to_vec(&series).expect("failed to serialize series");
-        debug!("Sending body: {:?}", &series);
-
+    pub async fn ship_series(&self, series: &Series) -> Result<Response, Error> {
         let url = format!("{}/api/v2/series", &self.metrics_intake_url_prefix);
-        let resp = self
-            .client
-            .post(&url)
-            .header("DD-API-KEY", &self.api_key)
-            .header("Content-Type", "application/json")
-            .body(body)
-            .send()
-            .await;
+        debug!("Sending body: {:?}", &series);
+        let body = serde_json::to_vec(&series).expect("failed to serialize series");
```

Comment:
```suggestion
        let body = serde_json::to_vec(&series).context("failed to serialize series")?;
```

Lets avoid the use of panic as much as possible.

Reply:
I just moved the old behavior, but agree, I will handle the error safely